hooks/pre/31-mongo_storage_engine.rb

```diff
@@ -0,0 +1,53 @@
+pulp_db_param = param('katello', 'pulp_db_name') || param('foreman_proxy_content', 'pulp_db_name')
+fail_and_exit('Unknown pulp database name') unless pulp_db_param
+MONGO_DB_NAME = pulp_db_param.value
+
+def fail_and_exit(message)
+  Kafo::Helpers.log_and_say :error, message
+  kafo.class.exit 1
+end
+
+def migration
+  export_dir = '/var/tmp/engine_upgrade'
+  mongo_dir = '/var/lib/mongodb'
+  hiera_file = '/etc/foreman-installer/custom-hiera.yaml'
+  mongo_backup = '/var/tmp/mongo_backup'
+
+  # Create export directory and dump MongoDB
+  logger.info 'Stopping Pulp services except MongoDB'
+  Kafo::Helpers.execute('katello-service stop --exclude rh-mongodb34-mongod')
+  Dir.mkdir_p(export_dir)
+  File.chmod(0700, export_dir)
+  logger.info "Starting mongodump to #{export_dir}"
+  Kafo::Helpers.execute("mongodump --host localhost --out #{export_dir}")
+
+  # Move datafiles out of MongoDB directory so it will start
+  logger.info 'Export done, stopping MongoDB to remove old datafiles'
+  Kafo::Helpers.execute('systemctl stop rh-mongodb34-mongod')
```

Comment:
IMHO we should avoid `katello-service` when possible. It's a katello specific thing and in this case `systemctl` provides the exact same features. The only thing it now adds is a dependency and overhead when you don't need the ordering.

Reply:
I tend to disagree. To me, this leads to inconsistent behavior across where we start and stop applications. Secondly, if say we put an improvement into `katello-service` targeted at a particular service and we've not used it everywhere then we are potentially introducing bugs. IMHO we should either use it everywhere or nowhere. Soon it will be `foreman-maintain` doing the service handling which isn't a katello specific thing :)

Reply:
I personally still think that we shouldn't rely on foreman-maintain for service handling. We just got rid of service_wait and personally I still like to continue that trend ;)